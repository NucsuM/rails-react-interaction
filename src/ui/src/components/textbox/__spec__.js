import React from "react";
import Enzyme, { shallow, mount, configure } from "enzyme";
import Textbox from "./Textbox";
import Adapter from 'enzyme-adapter-react-16';
import ToolBar from "../toolbar/Toolbar";

Enzyme.configure({ adapter: new Adapter() });

describe("Textbox", () => {
  let wrapper;
  const label = 'a_label';
  const text = 'typed_text';
  
  beforeEach(() => {
    wrapper = shallow(<Textbox name={label} onChange={text} />);
  });

  test("display the added text", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("it have the correct properties", () => {
    const label = wrapper.find('label')
    const input = wrapper.find('input')
    
    expect(label.props().children).toEqual('a_label:')
    expect(input.props().onChange).toEqual('typed_text')
  });
});
