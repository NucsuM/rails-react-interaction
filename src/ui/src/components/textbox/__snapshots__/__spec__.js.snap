// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox display the added text 1`] = `
<div>
  <label>
    a_label:
  </label>
  <input
    id="tb-id"
    name="tb-name"
    onChange="typed_text"
    type="text"
  />
</div>
`;
